src/Storages/StorageURL.cpp

```diff
@@ -151,8 +151,11 @@ namespace
                             std::size_t n = user_info.find(':');
                             if (n != std::string::npos)
                             {
+                                std::string urlEncodedPassword;
                                 credentials.setUsername(user_info.substr(0, n));
```

Comment:
Shouldn't we do the same percent decoding for the username also? https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1

Reply:
Sure! Actually I did this several days ago. But remove url decoding username at last since not very familiar with rfc3986. I decode the whole user_info string this time.

Reply:
```
2022-01-05 14:46:56 /usr/share/clickhouse-test/queries/0_stateless/02157_url_auth.sh: line 39: docker: command not found
2022-01-05 14:46:56 /usr/share/clickhouse-test/queries/0_stateless/02157_url_auth.sh: line 40: docker: command not found
2022-01-05 14:46:56 /usr/share/clickhouse-test/queries/0_stateless/02157_url_auth.sh: line 45: docker: command not found
2022-01-05 14:46:56 /usr/share/clickhouse-test/queries/0_stateless/02157_url_auth.sh: line 46: docker: command not found
2022-01-05 14:46:56 
2022-01-05 14:46:56 stdout:
2022-01-05 14:46:56 Code: 1000. DB::Exception: Received from localhost:9000. DB::Exception: Connection refused. (POCO_EXCEPTION)
2022-01-05 14:46:56 Code: 1000. DB::Exception: Received from localhost:9000. DB::Exception: Connection refused. (POCO_EXCEPTION)
2022-01-05 14:46:56 Code: 1000. DB::Exception: Received from localhost:9000. DB::Exception: Connection refused. (POCO_EXCEPTION)
2022-01-05 14:46:56 Code: 1000. DB::Exception: Received from localhost:9000. DB::Exception: Connection refused. (POCO_EXCEPTION)
2022-01-05 14:46:56 
```
I found docker command is not supported any longer, that caused my failure.

Reply:
I've replaced with python test